src/lib/storage/chunk.cpp

```diff
@@ -1,31 +1,51 @@
 #include "chunk.hpp"
 
-#include "abstract_segment.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  _columns.push_back(segment);
+}
+
+template <typename T>
+bool append_with_type(std::vector<std::shared_ptr<AbstractSegment>>::iterator column_it,
+                      std::vector<AllTypeVariant>::const_iterator value_it) {
+  auto segment = std::dynamic_pointer_cast<ValueSegment<T>>(*column_it);
+  if (segment) {
+    segment->append(*value_it);
+    return true;
+  }
+  return false;
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == column_count(), "Cannot insert a tuple with less values than columns.");
+
+  auto column_it = _columns.begin();
+  auto value_it = values.begin();
+
+  for (; column_it < _columns.end(); ++column_it) {
```

Comment:
```suggestion
  for (const auto& column : _columns) {
```

Reply:
In principal, a good suggestion, but I'd like to keep to code more concise, when sticking with either iterators or foreach-es. Sadly, I do not see a possibility to avoid the value-iterator.